drivers/storage/portworx/util/util.go

```diff
@@ -955,29 +953,26 @@ func SecurityEnabled(cluster *corev1.StorageCluster) bool {
 }
 
 // SetupContextWithToken Gets token or from secret for authenticating with the SDK server
-func SetupContextWithToken(ctx context.Context, cluster *corev1.StorageCluster, k8sClient client.Client, skipSecurityCheck bool) (context.Context, error) {
-	// auth not declared in cluster spec
-	// if security enabled check is skipped, proceed without returning context
-	if !SecurityEnabled(cluster) && !skipSecurityCheck {
-		return ctx, nil
-	}
+func SetupContextWithToken(ctx context.Context, cluster *corev1.StorageCluster, k8sClient client.Client) (context.Context, error) {
 
+	// Get the secret key for the system issuer
+	// return error if error is not NotFound
 	pxAppsSecret, err := GetSecretValue(ctx, cluster, k8sClient, SecurityPXSystemSecretsSecretName, SecurityAppsSecretKey)
-	if err != nil {
-		if !skipSecurityCheck {
-			return ctx, fmt.Errorf("failed to get portworx apps secret: %v", err.Error())
-		}
-		// if security enabled check is skipped and secret is not present, proceed with dummy secret value
-		// this is case of fresh install where security was never enabled
-		if errors.IsNotFound(err) && skipSecurityCheck {
-			pxAppsSecret = DummySecretValue
-		}
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, err
 	}
 
-	if pxAppsSecret == "" {
+	// if secret is not found, check if security is enabled
+	if errors.IsNotFound(err) {
+		if SecurityEnabled(cluster) {
+			// when security is enabled in STC, and secret is not found, return error
+			return nil, fmt.Errorf("failed to get portworx apps secret: %v", err.Error())
```

Comment:
nit: include namespace/name of the secret in the error message
Also, use `("...: %w", err)`

Reply:
Just wondering why is this secret called apps secret? Shouldn't it just be a security token secret or security context secret?

Reply:
Since we are specifically trying to fetch **apps-secret** key here inside **px-system-secrets** secret, so the error message is specifying that.